src/SpeedyTransforms/spectral_transform.jl

```diff
@@ -337,102 +324,108 @@ function transform!(                # SPECTRAL TO GRID
+                gnorth[:, k, j] .*= coslat⁻¹[j]     # scale in place
+                gsouth[:, k, j] .*= coslat⁻¹[j]
             end
+        end
+    end
 
-            # INVERSE FOURIER TRANSFORM in zonal direction
-            brfft_plan = brfft_plans[j_north]       # FFT planned wrt nlon on ring
-            ilons = rings[j_north]                  # in-ring indices northern ring
-            LinearAlgebra.mul!(view(grids.data, ilons, k_grid), brfft_plan, view(gn, 1:nfreq))  # perform FFT
-
-            # southern latitude, don't call redundant 2nd fft if ring is on equator 
-            ilons = rings[j_south]                  # in-ring indices southern ring
-            not_equator && LinearAlgebra.mul!(view(grids.data, ilons, k_grid), brfft_plan, view(gs, 1:nfreq))  # perform FFT
+    # INVERSE FOURIER TRANSFORM in zonal direction (batched in the vertical)
+    rings = eachring(grids)                 # precomputed ring indices
 
-            fill!(gn, zero(Complex{NF}))            # set phase factors back to zero
-            fill!(gs, zero(Complex{NF}))
+    @inbounds for j_north in 1:nlat_half    # symmetry: loop over northern latitudes only
+        j = j_north                         # symmetric index / ring-away from pole index
+        j_south = nlat - j_north + 1        # southern latitude index
+        nlon = nlons[j]                     # number of longitudes on this ring (north or south)
+        nfreq  = nlon÷2 + 1                 # linear max Fourier frequency wrt to nlon
+        not_equator = j_north != j_south    # is the latitude ring not on equator?
+
+        brfft_plan = brfft_plans[j]         # FFT planned wrt nlon on ring
+        ilons = rings[j_north]              # in-ring indices northern ring
+
+        # TODO `out` currently needed as FFT requires stride 1 output
+        out = S.scratch_memory_grid[j]      # output array
+        LinearAlgebra.mul!(out, brfft_plan, view(gnorth, 1:nfreq, :, j))        # perform FFT
+        grids[ilons, :] .= out
+
+        # southern latitude, don't call redundant 2nd fft if ring is on equator 
+        ilons = rings[j_south]              # in-ring indices southern ring
+        if not_equator
+            LinearAlgebra.mul!(out, brfft_plan, view(gsouth, 1:nfreq, :, j))    # perform FFT
+            grids[ilons, :] .= out
         end
     end
 
     return grids
 end
 
+@inline function fused_oddeven_dot(a::AbstractVector, b::AbstractVector)
+    odd  = zero(eltype(a))      # dot prodcut with elements 1, 3, 5, ... of a, b
+    even = zero(eltype(a))      # dot product with elements 2, 4, 6, ... of a, b
+    
+    n = length(a)
+    n_even = n - isodd(n)       # if n is odd do last odd element after the loop
+    
+    @inbounds for i in 1:2:n_even
+        odd = muladd(a[i], b[i], odd)
+        even = muladd(a[i+1], b[i+1], even)
+    end
+    
+    # now do the last element if n is odd
+    odd = isodd(n) ? muladd(a[end], b[end], odd) : odd
+    return odd, even
+end
```

Comment:
Nitpicking: since you're taking an `AbstractVector` in input you shouldn't assume 1-based indexing, especially combined with `@inbounds`, however you can use [`Base.require_one_based_indexing`](https://docs.julialang.org/en/v1/devdocs/offset-arrays/#man-custom-indices) to ensure that, for example by adding
```
Base.require_one_based_indexing(a, b)
```
at the top of the function.  This check can be elided by the compiler if it can easily prove it's `true` (which is the case for standard `Array`)

Reply:
Added that line!